backend/tournesol/views/inconsistencies.py

```diff
@@ -0,0 +1,213 @@
+                                                                          uncertainty)
+
+            count_comparisons_analysed += 1
+            if inconsistency >= threshold:
+                inconsistency_sum += inconsistency
+
+                inconsistent_criterion_comparisons.append(
+                    {
+                        "inconsistency": inconsistency,
+                        "entity_1_uid": entity_1,
+                        "entity_2_uid": entity_2,
+                        "criteria": criterion,
+                        "comparison_score": comparison_score,
+                        "entity_1_rating": entity_1_score,
+                        "entity_2_rating": entity_2_score,
+                    }
+                )
+
+        mean_inconsistency = 0
+        if count_comparisons_analysed > 0:
+            mean_inconsistency = inconsistency_sum / count_comparisons_analysed
+
+        inconsistent_criterion_comparisons.sort(key=lambda d: d['inconsistency'], reverse=True)
+
+        # No need to return everything, return at most 100 elements
+        inconsistent_criterion_comparisons = inconsistent_criterion_comparisons[:100]
+
+        response = {
+            "mean_inconsistency": mean_inconsistency,
+            "count": len(inconsistent_criterion_comparisons),
+            "results": inconsistent_criterion_comparisons,
+        }
+
+        return response
+
+    @staticmethod
+    def _calculate_inconsistency(entity_1_calculated_rating,
+                                 entity_2_calculated_rating,
+                                 comparison_score,
+                                 uncertainty) -> float:
+        """
+        Calculate the inconsistency between the comparison
+        criterion score and the general rating of the entity.
+
+        Let's note R the rating difference (rating2 - rating1),
+        r its variable counterpart, U the uncertainty and C the
+        comparison (positive if the 2nd entity is preferred).
+        Consider this function i:
+        i(r) = |C - 10r / sqrt(r² + 1)| (cf issue #671 discussion)
+        The inconsistency is the minimum of i(r) for r in [R - U, R + U]
+
+        The function i(r), is either:
+        1 - always increasing, if c <= -10 (normally, c is in [-10, 10])
+        2 - decreasing to 0, then increasing (because it is an absolute value)
+        3 - always decreasing, if c >= 10
+        So, the inconsistency is either i(R - U) in case 1, inconsistency(R + U)
+        in case 3, or 0 in case 2.
+        The value of R for which i(r) = 0 is C / sqrt(100 - C²), which
+        helps in case 2 to know if the minimum is 0 by checking if this root
+        is in [R - U, R + U].
+
+        There is also an imprecision of 0.5 on the comparison,
+        since the comparisons are made on floats and not integers,
+        which is subtracted on the result of the previous calculation.
```

Comment:
If you have a real preference of, say, 3.5001, you can chose the set a comparison score of 4. So, there is at most a 0.5 imprecision in the comparison score if it's rounded to the nearest integer.

The comparison_score has a linear effect on the inconsistency. Adding 0.5 to the comparison_score will either increase the inconsistency by 0.5, or decrease it by 0.5. But since the imprecision could be both ways and we want to minimize the inconsistency, 0.5 is subtracted to the inconsistency.

Maybe this needs to be rephrased.

Reply:
In case someone is interested by how the inconsistency function varies **with the rating_difference**, here is a curve with uncertainty = 0 and comparison_score = -5
![inconsistency](https://user-images.githubusercontent.com/47578089/163873663-4219c592-6e5c-461b-bebf-a2456dbc4f8d.png)
And here is how it would look like if there weren't the absolute value (and the imprecision of 0.5)
![inconsistency_no_abs](https://user-images.githubusercontent.com/47578089/163874333-ac07df9f-b5d2-4f09-af7d-6cec56e41b30.png)